src/monitor.cpp

```diff
@@ -160,108 +160,139 @@ complex<double> fields_chunk::get_field(component c, const ivec &iloc) const {
 }
 
-double fields::get_eps(const vec &loc) const {
+double fields::get_eps(const vec &loc, double omega) const {
   double tr = 0;
   int nc = 0;
   FOR_ELECTRIC_COMPONENTS(c) {
     if (gv.has_field(c)) {
-      tr += get_chi1inv(c, component_direction(c), loc, false);
+      tr += get_chi1inv(c, component_direction(c), loc, omega, false);
       ++nc;
     }
   }
   return nc / sum_to_all(tr);
 }
 
-double fields::get_mu(const vec &loc) const {
+double fields::get_mu(const vec &loc, double omega) const {
   double tr = 0;
   int nc = 0;
   FOR_MAGNETIC_COMPONENTS(c) {
     if (gv.has_field(c)) {
-      tr += get_chi1inv(c, component_direction(c), loc, false);
+      tr += get_chi1inv(c, component_direction(c), loc, omega, false);
       ++nc;
     }
   }
   return nc / sum_to_all(tr);
 }
 
-double structure::get_chi1inv(component c, direction d, const ivec &origloc, bool parallel) const {
+double structure::get_chi1inv(component c, direction d, const ivec &origloc, double omega, bool parallel) const {
   ivec iloc = origloc;
   for (int sn = 0; sn < S.multiplicity(); sn++)
     for (int i = 0; i < num_chunks; i++)
       if (chunks[i]->gv.owns(S.transform(iloc, sn))) {
         signed_direction ds = S.transform(d, sn);
-        double val = chunks[i]->get_chi1inv(S.transform(c, sn), ds.d, S.transform(iloc, sn)) *
+        double val = chunks[i]->get_chi1inv(S.transform(c, sn), ds.d, S.transform(iloc, sn), omega) *
                      (ds.flipped ^ S.transform(component_direction(c), sn).flipped ? -1 : 1);
         return parallel ? sum_to_all(val) : val;
       }
   return 0.0;
 }
+// Useful if you already know the exact location of the point you are interested in
+// (i.e. you know what idx should be). This is used with the get_array() routines.
+double structure_chunk::get_chi1inv_at_pt(component c, direction d, int idx, double omega) const {
+  double res = 0.0;
+  if (is_mine()){
+    res =
+        chi1inv[c][d] ? chi1inv[c][d][idx] : (d == component_direction(c) ? 1.0 : 0);
+    if (res != 0){
+      // Get instaneous dielectric (epsilon)
+      std::complex<double> eps(1.0 / res,0.0);
+      // Loop through and add up susceptibility contributions
+      // locate correct susceptibility list
+      susceptibility *Esus = chiP[E_stuff];
+      while (Esus) {
+        if (Esus->sigma[c][d]) {
+          double sigma = Esus->sigma[c][d][idx];
+          eps += Esus->chi1(omega,sigma);
+        } 
+        Esus = Esus->next;
+      }
+      // Account for conductivity term
+       if (conductivity[c][d]) {
+         double conductivityCur = conductivity[c][d][idx];
+         eps = std::complex<double>(1.0, (conductivityCur/omega)) * eps;
+       }
+      // Return chi1 inverse, take the real part since no support for loss in mpb yet
+      // TODO: Add support for metals
+      if (eps.imag() == 0.0){
+        res = 1.0 / eps.real();
+      }else{
+        res = 1.0 / (std::sqrt(eps).real() * std::sqrt(eps).real());
```

Comment:
What's going on here?   Why not just `res = real(1.0 / eps)`?

Reply:
What we really want is `real(1/n) ^2`, which is not the same as `real(1.0 / eps)` if `eps` is complex.